pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,264 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Implementation of the BaseRepository abstract classes that defines an in-memory
+repository based on dictionaries.
+"""
+
+from __future__ import absolute_import, print_function
+
+import six
+
+from pywbem import CIMClass, CIMQualifierDeclaration, CIMInstance
+
+from pywbem._nocasedict import NocaseDict
+from pywbem._utils import _format
+
+from ._baserepository import BaseObjectStore, BaseRepository
+from ._utils import _uprint
+
+__all__ = ['InMemoryRepository', 'InMemoryObjectStore']
+
+
+class InMemoryObjectStore(BaseObjectStore):
+    """
+    Derived from the :class:`BaseObjectStore`, this class implements a
+    dictionary based in-memory repository for CIM objects that manages
+    CIM classes, CIM instances, and CIM qualifier declarations.
```

Comment:
Let's not use the term "repository" for the object store. Also, since this sentence is used as a brief description, it can be simplified. Suggest this for the first sentence:
```
In-memory store for CIM objects of a single type (CIM classes, CIM instances, or CIM qualifier declarations).
```

Reply:
DONE